torchx/runner/api.py

```diff
@@ -360,25 +360,13 @@ def dryrun(
             resolved_cfg = sched.run_opts().resolve(cfg)
             if workspace and isinstance(sched, WorkspaceMixin):
                 role = app.roles[0]
-                old_img = role.image
 
                 logger.info(f"Checking for changes in workspace `{workspace}`...")
                 logger.info(
                     'To disable workspaces pass: --workspace="" from CLI or workspace=None programmatically.'
                 )
                 sched.build_workspace_and_update_role(role, workspace, resolved_cfg)
 
-                if old_img != role.image:
```

Comment:
moving this logging message affects users who are using other `workspace` implementations.